apps/landing-page/pages/blog/exploring-quay-registry-in-backstage.mdx

```diff
@@ -0,0 +1,116 @@
+
+Backstage charts and models the software ecosystem as Catalog entities. Users compose small individual service components into a bigger picture. And in order to truly understand and fully describe the individual building blocks, Backstage users construct views to capture different aspects of these components. From technical documentation, through dependencies and relation to deployment state, CI and CD. And part of this picture is understanding what is actually being deployed into live environments: a view of the available and deployed container images. This new Quay plugin for Backstage brings that view.
+
+Quay is an OCI-compatible registry that allows users to build, store and distribute container images and other OCI artifacts. It is available as a hosted service on quay.io as well as a self-hosted environment deployable to any OpenShift cluster through a Quay operator. 
+
+
+## Installation and setup
+
+With this plugin, viewing available container images in a particular repository is easy. The following guide will elaborate in detail on individual steps and all currently supported settings and options.
+
+First, it is necessary to install the plugin. Please add it to the frontend of your Backstage instance:
+
+```bash
+yarn workspace app add @janus-idp/backstage-plugin-quay
+```
+
+### Connecting to Quay registry via Proxy
+
+Plugin leverages the Backstage native proxy capabilities to query Quay API; therefore, it needs to be configured in `app-config.yaml`. In order to connect the Backstage instance to the public hosted Quay.io environment, the following configuration can be used:
+
+```yaml filename="app-config.yaml"
+proxy:
+  '/quay/api':
+    target: 'https://quay.io'
+    changeOrigin: true
+    headers:
+      X-Requested-With: 'XMLHttpRequest'
+```
+
+When accessing private Quay repositories, it is necessary to extend this configuration with `Authorization` header and Quay API token. This token can be obtained by creating a Quay Application using the [steps outlined in this documentation](https://docs.quay.io/api/#applications-and-tokens). Once a token is obtained, the header can be set by extending the `app-config.yaml` setup above with an additional header:
+
+```yaml {7} filename="app-config.yaml"
+proxy:
+  '/quay/api':
+    target: 'https://quay.io'
+    changeOrigin: true
+    headers:
+      X-Requested-With: 'XMLHttpRequest'
+      Authorization: 'Bearer ${QUAY_TOKEN}'
+```
+
+Be aware that the `QUAY_TOKEN` is an environment variable that has to be available to the Backstage instance at runtime.
+
+Another popular demand may be to use this plugin to access a private registry, a self-hosted Quay deployment. This can be achieved by simply changing the `target` property in the settings above. In addition to that, if the self-hosted Quay registry is also deployed with a self-signed certificate that is not in the certificate chain of trust for the Backstage instance, the `secure` option has to be unset.
```

Comment:
```suggestion
Another popular option is to be able to target a self-hosted Quay deployment. This can be achieved by simply changing the `target` property in the settings above with the location of the Quay instance. In addition, if the self-hosted Quay registry is also deployed with a certificate that is not in the certificate chain of trust for the Backstage instance, the `secure` option has to be unset.
```

Reply:
Would be a good feature of the plugin to support specifying the location of a certificate that can be used to support a non-public CA

Reply:
It's using https://www.npmjs.com/package/http-proxy-middleware behind the scenes. It seems like it doesn't allow you to specify custom CA in any way. But you may be able to mount it to the container maybe? Anyway we may want to explore that in future.